front/app/modules/commercial/insights/api/inputs/useInputs.ts

```diff
@@ -0,0 +1,42 @@
+import { useQueryClient, useQuery } from '@tanstack/react-query';
+import { CLErrors } from 'typings';
+import fetcher from 'utils/cl-react-query/fetcher';
+import inputsKeys from './keys';
+import { IInsightsInputs, QueryParameters, InputsKeys } from './types';
+
+export const defaultPageSize = 20;
+
+const fetchInputs = (
+  viewId: string,
+  { pageNumber, pageSize, category, ...queryParams }: QueryParameters
+) =>
+  fetcher<IInsightsInputs>({
+    path: `/insights/views/${viewId}/inputs`,
+    action: 'get',
+    queryParams: {
+      'page[number]': pageNumber || 1,
+      'page[size]': pageSize || defaultPageSize,
+      categories: typeof category === 'string' ? [category] : undefined,
+      ...queryParams,
+    },
+  });
+
+const useInputs = (viewId: string, queryParams: QueryParameters) => {
+  const queryClient = useQueryClient();
+
+  const prefetchParams = {
```

Comment:
Prefetching the next page in a table example. This way, the user has the data available and does not need to wait for it when they switch to the next page.